sqlglot/expressions.py

```diff
@@ -535,6 +535,35 @@ def assert_is(self, type_):
         assert isinstance(self, type_)
         return self
 
+    def validate(self, args: t.Optional[t.List] = None) -> t.Generator[str, None, None]:
```

Comment:
why is args optional?

Reply:
don't yield, just return a list of error messages?

Reply:
Because we might've instantiated a function with kwargs, or the args used @ instantiation time might not be known at all @ validation time.

Reply:
the name validate is not good if you're going to keep it a generator, it should be like error_messages

Reply:
Changed